addons/account/models/account_move_line.py

```diff
@@ -662,6 +663,26 @@ def _search_account_id(self, operator, value):
 
         return [('account_id', operator, value)]
 
+    @api.model
+    def _search_partner_id(self, operator, value):
+        if (
+            operator in ('in', 'not in', 'any', 'not any')
+            and not isinstance(value, (tuple, list, OrderedSet))
```

Comment:
I know you've inspired yourself from `search_account_id`, but it didn't take into account doing `ilike`, so in the case of your patch, the search method doesn't do anything, since the `operator` is `ilike`. You may need to bifurcate behaviour based on the operator.

Reply:
Ok, I didn't know that it was transformed before the search invocation, tnx for letting me know.
That simplifies the implementation.
The patch looks ready to me, tomorrow I will do some benchmarks and add them to the PR description. :)